### PR TITLE
ChatController Testing

### PR DESCRIPTION
**Pull request by plutowootoe**

### CODE REVIEW

**Review by Bojo992 — CHANGES_REQUESTED** [review]
Can you change which brunch you are pushing to dev, I will take a look at the tests later today, sorry just seen pr.

**Review by oakwoodwolf — APPROVED** [review]
lgtm

**Review by Bojo992 — CHANGES_REQUESTED** [review]
I think you would need to add Testcontainer for MySql

File: project/src/test/java/com/cs4337/project/ChatControllerTests.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    public void sendTestMessage(StompSession session) throws IOException {
+
+        ChatMessage testMessage = new ChatMessage(
+                "Test message",           // content
+                "testUser",                // sender
+                MessageType.CHAT, // type
+                LocalDateTime.now(),       // sentAt
+                null,                      // media (null if not needed)
+                false                      // isSeen (false when first sent)
+        );
```
Comment (Bojo992) [inline project/src/test/java/com/cs4337/project/ChatControllerTests.java]: There is better way to make is with builder

**Comment by plutowootoe** [comment]
> Can you change which brunch you are pushing to dev, I will take a look at the tests later today, sorry just seen pr.

Added changes

**Review by oakwoodwolf — APPROVED** [review]

**Review by Bojo992 — CHANGES_REQUESTED** [review]
You would need to add something to mock kafka in bouth of test files.

File: project/src/test/java/com/cs4337/project/ChatControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
```
Comment (Bojo992) [inline project/src/test/java/com/cs4337/project/ChatControllerTests.java:42]: You would need to add a variable that would represent random port

**Review by Bojo992 — APPROVED** [review]
it's-done-ray-johnson

**Pull request merged**